src/components/FormAlertWithSubmitButton.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import {StyleProp, View, ViewStyle} from 'react-native';
+import useThemeStyles from '@styles/useThemeStyles';
+import Button from './Button';
+import FormAlertWrapper from './FormAlertWrapper';
+
+type FormAlertWithSubmitButtonProps = {
+    /** Error message to display above button */
+    message?: string;
+
+    /** Whether the button is disabled */
+    isDisabled?: boolean;
+
+    /** Whether message is in html format */
+    isMessageHtml?: boolean;
+
+    /** Styles for container element */
+    containerStyles?: StyleProp<ViewStyle>;
+
+    /** Is the button in a loading state */
+    isLoading?: boolean;
+
+    /** Callback fired when the "fix the errors" link is pressed */
+    onFixTheErrorsLinkPressed?: () => void;
+
+    /** Submit function */
+    onSubmit: () => void;
+
+    /** Should the button be enabled when offline */
+    enabledWhenOffline?: boolean;
+
+    /** Disable press on enter for submit button */
+    disablePressOnEnter?: boolean;
+
+    /** Whether the form submit action is dangerous */
+    isSubmitActionDangerous?: boolean;
+
+    /** Custom content to display in the footer after submit button */
+    footerContent?: React.ReactNode;
+
+    /** Styles for the button */
+    buttonStyles?: StyleProp<ViewStyle>;
+
+    /** Whether to show the alert text */
+    isAlertVisible: boolean;
+
+    /** Text for the button */
+    buttonText: string;
+};
+
+function FormAlertWithSubmitButton({
+    message = '',
+    isDisabled = false,
+    isMessageHtml = false,
+    containerStyles,
+    isLoading = false,
+    onFixTheErrorsLinkPressed = () => {},
+    enabledWhenOffline = false,
+    disablePressOnEnter = false,
+    isSubmitActionDangerous = false,
+    footerContent = null,
+    buttonStyles,
+    buttonText,
+    isAlertVisible,
+    onSubmit,
+}: FormAlertWithSubmitButtonProps) {
+    const styles = useThemeStyles();
+    const style = [!footerContent ? {} : styles.mb3, buttonStyles];
+
+    return (
+        <FormAlertWrapper
+            containerStyles={[styles.mh5, styles.mb5, styles.justifyContentEnd, containerStyles]}
```

Comment:
If `containerStyles` is an array, it needs to be spread here

Reply:
@pac-guerreiro It will work, look at screenshots
![image](https://github.com/Expensify/App/assets/25230417/669d2e45-0fdc-4103-850b-11d5c7dedda3)
<img width="354" alt="image" src="https://github.com/Expensify/App/assets/25230417/599cd78f-2db1-45a5-8060-d41d0bd4e0c8">

Reply:
The style props are able to handle nested arrays 🙂

Reply:
Nvm then @kubabutkiewicz @fabioh8010, I had the idea that we could not pass nested arrays like that

Reply:
Sorry if I am being dumb, but I am seeing this console warning and I believe it is related to this change:
<img width="656" alt="image" src="https://github.com/Expensify/App/assets/5201282/07604c00-9dc0-422b-a4f9-93e2fec98bc9">